Library/Homebrew/api.rb

```diff
@@ -82,9 +83,8 @@ def self.fetch_json_api_file(endpoint, target:)
           opoo "#{target.basename}: update failed, falling back to cached version."
         end
 
-        FileUtils.touch target
         JSON.parse(target.read)
-      rescue JSON::ParserError
+      rescue JSON::ParserError, Errno::ENOENT
```

Comment:
How can we get to this point that we get `Errno::ENOENT`? We should have only skipped the download if it exists. If we tried the download and it failed we should be raising an exception or retrying above?